core/src/main/java/org/apache/calcite/plan/volcano/RelSet.java

```diff
@@ -413,14 +411,9 @@ void mergeWith(
     // calls propagateCostImprovements() for RelSubset instances,
     // whose best should be changed to check whether that
     // subset's parents get cheaper.
-    Set<RelSubset> activeSet = new HashSet<>();
     for (Map.Entry<RelSubset, RelNode> subsetBestPair : changedSubsets.entrySet()) {
-      RelSubset relSubset = subsetBestPair.getKey();
```

Comment:
The `changedSubsets` keys are never used.

Reply:
fixed